### ISSUE TITLE
Compilation error on WSL and Red Hat 

### ISSUE THREAD

**Issue by Joseph459459** [msoos/cryptominisat#766]
Hello! Thank you for this tool - I am very interested in the XOR clause feature.

I just cloned the code today and tried to build on WSL as well as my computing cluster which runs Red Hat and ran into the following error on both:
```
[  1%] Building CXX object cmsat5-src/CMakeFiles/cryptominisat5.dir/cnf.cpp.o
In file included from /home/joe459/cryptominisat/src/cnf.h:31:0,
                 from /home/joe459/cryptominisat/src/cnf.cpp:23:
/home/joe459/cryptominisat/src/vardata.h:36:13: error: expected unqualified-id before ‘[’ token
     VarData([[maybe_unused]] uint32_t num) {
             ^
/home/joe459/cryptominisat/src/vardata.h:36:13: error: expected ‘)’ before ‘[’ token
/home/joe459/cryptominisat/src/cnf.cpp: In member function ‘void CMSat::CNF::enlarge_nonminimial_datastructs(size_t)’:
/home/joe459/cryptominisat/src/cnf.cpp:137:49: error: no matching function for call to ‘CMSat::VarData::VarData(std::vector<CMSat::VarData>::size_type)’
         varData.push_back(VarData(varData.size()));
                                                 ^
In file included from /home/joe459/cryptominisat/src/cnf.h:31:0,
                 from /home/joe459/cryptominisat/src/cnf.cpp:23:
/home/joe459/cryptominisat/src/vardata.h:34:8: note: candidate: CMSat::VarData::VarData()
 struct VarData
        ^~~~~~~
/home/joe459/cryptominisat/src/vardata.h:34:8: note:   candidate expects 0 arguments, 1 provided
/home/joe459/cryptominisat/src/vardata.h:34:8: note: candidate: constexpr CMSat::VarData::VarData(const CMSat::VarData&)
/home/joe459/cryptominisat/src/vardata.h:34:8: note:   no known conversion for argument 1 from ‘std::vector<CMSat::VarData>::size_type {aka long unsigned int}’ to ‘const CMSat::VarData&’
/home/joe459/cryptominisat/src/vardata.h:34:8: note: candidate: constexpr CMSat::VarData::VarData(CMSat::VarData&&)
/home/joe459/cryptominisat/src/vardata.h:34:8: note:   no known conversion for argument 1 from ‘std::vector<CMSat::VarData>::size_type {aka long unsigned int}’ to ‘CMSat::VarData&&’
cmsat5-src/CMakeFiles/cryptominisat5.dir/build.make:75: recipe for target 'cmsat5-src/CMakeFiles/cryptominisat5.dir/cnf.cpp.o' failed
make[2]: *** [cmsat5-src/CMakeFiles/cryptominisat5.dir/cnf.cpp.o] Error 1
CMakeFiles/Makefile2:128: recipe for target 'cmsat5-src/CMakeFiles/cryptominisat5.dir/all' failed
make[1]: *** [cmsat5-src/CMakeFiles/cryptominisat5.dir/all] Error 2
Makefile:135: recipe for target 'all' failed
make: *** [all] Error 2
```

I basically copy/pasted the instructions on your README for the WSL (Ubuntu 18.04) build. For the Red Hat machine, I don't have installation privileges so may have some funny packages (and may be missing libgmp). I'm still surprised they give the same error and so I'm wondering if this is a source code error.

**Comment by msoos** [comment]
Ubuntu 18.04 is very old and will not work. You need a more modern compiler. I am closing, because only C++ versions 20 and above are supported. You need at least Ubuntu 22.04.  Once you get that, it will work. Otherwise, you can also use the statically compiled executable. It should work under WSL! See the "Releases" on the right on the GitHub page.

Cheers,

Mate

**Comment by Joseph459459** [comment]
Solved by updating to Ubuntu 24.04 for WSL! Thanks
